pallets/parachain-staking/src/vote.rs

```diff
@@ -0,0 +1,223 @@
+        if <ParachainStaking<T> as Store>::VotesRepository::contains_key(self.growth_id) {
+            return Ok(ParachainStaking::<T>::get_vote(self.growth_id))
+        }
+        return Err(DispatchError::Other("Growth data is not found in votes repository"))
+    }
+
+    fn is_valid(&self) -> bool {
+        return false;
+        let voting_session_data = self.state();
+        let growth_data_result = ParachainStaking::<T>::try_get_growth_data(&self.growth_id);
+        let growth_is_pending_approval =
+            <ParachainStaking<T> as Store>::PendingApproval::contains_key(&self.growth_id.period);
+        let voting_session_exists_for_growth =
+            <ParachainStaking<T> as Store>::VotesRepository::contains_key(&self.growth_id);
+
+        if growth_data_result.is_err() ||
+            !growth_is_pending_approval ||
+            !voting_session_exists_for_growth ||
+            voting_session_data.is_err()
+        {
+            return false
+        }
+
+        let growth_already_processed =
+            <ParachainStaking<T> as Store>::ProcessedGrowthPeriods::contains_key(&self.growth_id.period);
+        let growth_period_is_complete = 
+            self.growth_id.period < <ParachainStaking<T> as Store>::GrowthPeriod::get().index;
+
+        let pending_approval_growth_ingress_counter =
+            <ParachainStaking<T> as Store>::PendingApproval::get(self.growth_id.period);
+        let vote_is_for_correct_ingress_counter =
+            pending_approval_growth_ingress_counter == self.growth_id.ingress_counter;
+        
+        let voting_session_is_finalised =
+            AVN::<T>::is_block_finalised(voting_session_data.expect("checked").created_at_block);
+
+        return !growth_already_processed &&
+            growth_period_is_complete &&
+            vote_is_for_correct_ingress_counter &&
+            voting_session_is_finalised
+    }
+
+    fn is_active(&self) -> bool {
+        let voting_session_data = self.state();
+        return voting_session_data.is_ok() &&
+            <frame_system::Pallet<T>>::block_number() <
+                voting_session_data.expect("voting session data is ok").end_of_voting_period &&
+            self.is_valid()
+    }
+
+    fn record_approve_vote(
+        &self,
+        voter: T::AccountId,
+        approval_signature: ecdsa::Signature,
+    ) -> DispatchResult {
+        <ParachainStaking<T> as Store>::VotesRepository::try_mutate(
+            &self.growth_id,
+            |vote| -> DispatchResult {
+                vote.ayes.try_push(voter).map_err(|_| avn_error::<T>::VectorBoundsExceeded)?;
+                vote.confirmations
+                    .try_push(approval_signature)
+                    .map_err(|_| avn_error::<T>::VectorBoundsExceeded)?;
+                Ok(())
+            },
+        )?;
+        Ok(())
+    }
+
+    fn record_reject_vote(&self, voter: T::AccountId) -> DispatchResult {
```

Comment:
Why don't we use signatures with a reject vote, as we do with approvals?

Reply:
Because we don't need to send anything to Ethereum if we reject a proposal. The AvN signature is required because we need to make sure a genuine collator is voting but the Eth signature is not required.